docs/contributing/contributing-rules.md

```diff
@@ -0,0 +1,136 @@
+    - Select a rule from a category in **Semgrep Registry**. Modify the rule or test code, click **Save**, and then **Fork**.
+4. Click **Share**.
+
+This workflow automatically creates a pull request in the GitHub [rules repository](https://github.com/returntocorp/semgrep-rules). Find more about the rules repository by reading the [Rule writing](#rule-writing) and [Tests](#tests) sections.
+
+### Contributing through GitHub
+
+Fork our repository and make a pull request; we'll contact you about signing our Contributor License Agreement (CLA). Install pre-commit (see [installing pre-commit](#installing-pre-commit)) and make a pull request to the [rules repository](https://github.com/returntocorp/semgrep-rules) with two files:
+1. The semgrep pattern (.yml)
+2. The test file (with the file extension of the language or framework). The test file must contain at least one true positive and one true negative test case in order to get approved.
+
+See an example of a [pull request](https://github.com/returntocorp/semgrep-rules/pull/1728/files) to the rules repository. Pull requests require the approval of at least one maintainer and successfully passed [CI jobs](https://github.com/returntocorp/semgrep-rules/actions).
+
+Find more about the rules repository by reading the [Rule writing](#rule-writing) and [Tests](#tests) sections.
+
+#### Installing pre-commit
+
+Install pre-commit if you are contributing to the rules repository through GitHub but you are **not** contributing through sharing your rule in Semgrep App.
+
+1. Install [pre-commit](https://pre-commit.com/) by issuing the following command:
+    ```sh
+    python -m pip install pre-commit
+    ```
+2. Install the pre-commit hooks:
+    ```sh
+    pre-commit install
+    ```
+To check if `pre-commit` is working as expected, run the following command:
+    ```sh
+    pre-commit run --all
+    ```
+
+Once `pre-commit` is set up you may commit code and create pull requests.
+
+## Rule writing
+
+### Understanding rules repository file structure
+
+The namespacing format for contributing rules in the [rules repository](https://github.com/returntocorp/semgrep-rules) is `<language>/<framework>/<category>/$MORE`. If the rule does not belong to a particular framework, add it to the language directory, which uses the word `lang` in place of the `<framework>`.
+
+### Rule messages
+
+A well-written rule message includes:
+
+1. Description of the pattern. For example: missing parameter, dangerous flag, out-of-order function calls.
+2. Description of why this pattern was detected. For example: logic bug, introduces a security vulnerability, bad practice.
+3. An alternative that resolves the issue. For example: Use another function, validate data first, and discard the dangerous flag.
+
+For an example of a good rule message, see [this rule for Django's mark_safe()](https://github.com/returntocorp/semgrep-rules/blob/develop/python/django/security/audit/avoid-mark-safe.yaml).
+
+:::note
```

Comment:
This doesn't appear to be rendering correctly for me on GitHub. I checked it by clicking the three dots in the [PR review page](https://github.com/returntocorp/semgrep-docs/pull/449/files#diff-93a189106ae23d263b765c05700450812fe374b4f660c0ccddc1d550aad2439c) and clicking "View File".